### DISCUSSION TITLE
Publish module with Codeberg instead of Github

### DISCUSSION THREAD

**Discussion by hs0ucy** [keplerproject/luarocks#1522] [General]
Hi,

With Codeberg I can't use URI like, `git://codeberg.org/hs0ucy/Satelito.git` ... I only have access to `https://codeberg.org/hs0ucy/Satelito.git` or via SSH `git@codeberg.org:hs0ucy/Satelito.git`.

So I try with the https one. The upload went well.

But when It won't install:

```
luarocks install satelito --local 
Warning: falling back to wget - install luasec to get native HTTPS support
Installing https://luarocks.org/satelito-beta-3.src.rock

Error: Couldn't extract archive /tmp/luarocks_satelito-beta-3-8373946/Satelito.git: unrecognized filename extension
```

Maybe luarocks thinks it's a tarball?

So, in that case what should I do?

Thanks

**Comment by alerque** [comment]
Did you try `git+https://codeberg.org/hs0ucy/Satelito.git` as the url?

**Reply by hs0ucy** [reply]
It works! Thanks @alerque.

**Reply by alerque** [reply]
You're welcome. This is the correct way to enter the URL universally. The only reason it was working for GitHub with the other syntax is because the old URL scheme used to be supported there before they deprecated it and LuaRocks had to add special handling so old rockspecs that were valid at the time of publishing didn't need to be updated *en masse*.
